zap/src/main/java/org/parosproxy/paros/core/scanner/Alert.java

```diff
@@ -1277,10 +1284,26 @@ public final Alert build() {
             if (alertRef != null) {
                 alert.setAlertRef(alertRef);
             }
+            addCweTag(cweId);
             alert.setTags(tags);
```

Comment:
Not part of this PR but we should be setting a copy, otherwise changes done afterwards to the builder might affect an alert already built.

Reply:
I looked at this quickly, from my understanding it would need a deep copy which there doesn't seem to be a simple solution to. https://www.baeldung.com/java-copy-hashmap

I only checked that one article so I could have just missed the point.

Reply:
e.g. `new HashMap(tags)` would be enough to copy. Now that I'm reading the code again, it will have to be addressed in this PR anyway, to add the CWE tag.

Reply:
Okay.